sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1425,11 +1445,39 @@ class Analyzer(
       // rule: ResolveDeserializer.
       case plan if containsDeserializer(plan.expressions) => plan
 
-      // SPARK-25942: Resolves aggregate expressions with `AppendColumns`'s children, instead of
-      // `AppendColumns`, because `AppendColumns`'s serializer might produce conflict attribute
-      // names leading to ambiguous references exception.
-      case a @ Aggregate(groupingExprs, aggExprs, appendColumns: AppendColumns) =>
-        a.mapExpressions(resolveExpressionTopDown(_, appendColumns))
+      case a: Aggregate =>
```

Comment:
We need a high-level comment to explain why we trim alias here, e.g.
```
// SPARK-31670: ...
```

Reply:
> We need a high-level comment to explain why we trim alias here, e.g.
> 
> ```
> // [SPARK-31670](https://issues.apache.org/jira/browse/SPARK-31670): ...
> ```

Done